src/download.tsx

```diff
@@ -108,13 +108,30 @@ class Download extends KalturaPlayer.core.BasePlugin {
     );
   }
 
+  private shouldInjectUI(downloadMetadata: DownloadMetadata): boolean {
+    if (!downloadMetadata) return false;
+    const {flavors, captions, attachments, imageDownloadUrl} = downloadMetadata;
+    if (flavors.length || imageDownloadUrl) {
+      return true;
+    }
+    // getting here means that there are no flavors and there is no image (for example, a youtube entry with captions)
+    const {displayCaptions, displayAttachments} = this.downloadPluginManager.downloadPlugin.config;
+    if (captions.length && !attachments.length && !displayCaptions) {
```

Comment:
1. You also have displayFlavors param to manage
const shouldDisplay = 
displayFlavors && (flavors.length || imageDownloadUrl) ||
displayCaptions && captions.length ||
displayAttachments && attachments.length;

So for each case, it must be its flag && (existence of data)

Reply:
flavors is a different scenario. `displayFlavors` configuration, unlike `displayCaptions` and `displayAttachments`, is only controlling the "more flavors"; therefore, if i have flavors or image (which i check on line #114), i need to return true.

Reply:
and inside the sources-list component i manage `displayFlavors` configuration, which affects on whether to show all flavors and not just the default/configured one.
Also, as i wrote in my comment in line #117, if i get there, then flavors and image are both empty- there is no need to check them.